BrainPortal/app/models/cluster_task.rb

```diff
@@ -1687,6 +1689,151 @@ def docker_commands
+      taskfiles.each do |taskfile|
+        #self.addlog("Found new subtask file: #{taskfile}")
+        begin
+          file_content = File.read(taskfile)
+          self.submit_task_from_string(taskfile,file_content)
+        rescue => ex
+          self.addlog_exception(ex,"Error submitting subtask")
+        ensure
+          File.unlink(taskfile) rescue nil # to make sure we won't retry forever
+          #For debug
+          #File.rename(taskfile,"#{taskfile}.processed")
+        end
+      end # each new json descriptor
+      self.status_transition("Subtasking", current_status)
+    end
+  rescue  # TODO log error?
+    nil
+  end
+
+  protected
+
+  # Creates and submits a subtask defined by a JSON object.
+  # Parameters:
+  # * json_string: A string containing a JSON object defining the task.
+  #                Example:
+  #                  {
+  #                    "tool-class": "CbrainTask::TestTool",
+  #                    "description": "A task running TestTool",
+  #                    "parameters": {
+  #                       "important_number": "123",
+  #                       "dummy_parameter": "432"
+  #                    }
+  #                  }
+  def submit_task_from_string(json_filename,json_string)
+
+    # Parses JSON string and checks format
+    validate_json_string(json_string) # Raises an exception if string is not valid
+    new_task_hash = JSON.parse(json_string)
+
+    # Extract the main info about this new task
+    new_task_class_name  = new_task_hash["tool-class"].presence    || "ErrorNoSuchClassMyFriend"
+    new_share_wd_tid     = new_task_hash["share_wd_tid"].presence     # optional
+    new_tcid             = new_task_hash["tool-config-id"].presence   # optional
+    new_params           = new_task_hash["parameters"].presence    || {}
+    new_description      = new_task_hash["description"].presence   || "Task submitted by task #{self.id}"
+    new_prereqs          = new_task_hash["prerequisites"].presence || ""
+
+    # Prints log message in the task's logs
+    self.addlog("Submitting new subtask #{new_task_class_name}")
+
+    # Creates task
+    new_task              = CbrainTask.const_get(new_task_class_name).new # Raises an exception if tool class is not found
+    raise "Invalid tool class: #{new_task_class_name}" unless new_task.class < ClusterTask
+
+    new_task.batch_id     = self.id
+    new_task.launch_time  = Time.now
+    self.level            = 0 if self.level.blank?
+    new_task.level        = self.level + 1 # New task will be one level up its "parent" task in the task table.
+    new_task.share_workdir_with(new_share_wd_tid, "Queued") if new_share_wd_tid # can be nil
+
+    # Sets tool config among tool configs accessible by user of current task
+    new_tool                = new_task.tool
+    accessible_tool_configs = ToolConfig.find_all_accessible_by_user(self.user).where(:tool_id => new_tool.id, :bourreau_id => self.bourreau_id)
```

Comment:
Okay, when we need to submit to other bourreaux we'll easily find the code and modify it accordingly. For the moment the default provides a bit of security.